### ISSUE TITLE
conda install seaborn error on linux

### ISSUE THREAD

**Issue by DaanVanHauwermeiren** [conda/conda#3661]
got the following error when trying to install the seaborn package (v0.7.0) for a user:

```
conda install seaborn=0.7.0
Fetching package metadata: ....
Solving package specifications: .........................................
Package plan for installation in environment /home/data/smortier/.conda/envs/PBM:

The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    libgfortran-3.0.0          |                1         281 KB
    mkl-11.3.3                 |                0       122.1 MB
    pixman-0.32.6              |                0         2.4 MB
    ------------------------------------------------------------
                                           Total:       124.8 MB

The following NEW packages will be INSTALLED:

    cycler:           0.10.0-py27_0      (soft-link)
    dbus:             1.10.10-0          (soft-link)
    expat:            2.1.0-0            (soft-link)
    glib:             2.43.0-1           (soft-link)
    gst-plugins-base: 1.8.0-0            (soft-link)
    gstreamer:        1.8.0-0            (soft-link)
    icu:              54.1-0             (soft-link)
    libffi:           3.2.1-0            (soft-link)
    libgcc:           5.2.0-0            (soft-link)
    libxcb:           1.12-0             (soft-link)
    mkl:              11.3.3-0          
    pycairo:          1.10.0-py27_0      (soft-link)
    seaborn:          0.7.0-py27_0       (soft-link)

The following packages will be UPDATED:

    cairo:            1.12.18-3          --> 1.12.18-6          (soft-link)
    fontconfig:       2.11.1-3           --> 2.11.1-6           (soft-link)
    freetype:         2.5.2-0            --> 2.5.5-1            (soft-link)
    jpeg:             8d-0               --> 8d-2               (soft-link)
    libgfortran:      1.0-0              --> 3.0.0-1           
    libpng:           1.5.13-1           --> 1.6.22-0           (soft-link)
    libxml2:          2.9.0-0            --> 2.9.2-0            (soft-link)
    matplotlib:       1.4.3-np19py27_1   --> 1.5.3-np111py27_0  (soft-link)
    numpy:            1.10.1-py27_0      --> 1.11.2-py27_0      (soft-link)
    openssl:          1.0.2d-0           --> 1.0.2j-0           (soft-link)
    pandas:           0.17.0-np110py27_0 --> 0.19.0-np111py27_0 (soft-link)
    pip:              7.1.2-py27_0       --> 8.1.2-py27_0       (soft-link)
    pixman:           0.26.2-0           --> 0.32.6-0          
    pyparsing:        2.0.3-py27_0       --> 2.1.4-py27_0       (soft-link)
    pyqt:             4.11.3-py27_1      --> 5.6.0-py27_0       (soft-link)
    python:           2.7.10-2           --> 2.7.12-1           (soft-link)
    python-dateutil:  2.4.2-py27_0       --> 2.5.3-py27_0       (soft-link)
    pytz:             2015.7-py27_0      --> 2016.7-py27_0      (soft-link)
    qt:               4.8.6-1            --> 5.6.0-1            (soft-link)
    scipy:            0.15.1-np19py27_0  --> 0.18.1-np111py27_0 (soft-link)
    setuptools:       18.4-py27_0        --> 27.2.0-py27_0      (soft-link)
    sip:              4.16.5-py27_0      --> 4.18-py27_0        (soft-link)
    sqlite:           3.8.4.1-1          --> 3.13.0-0           (soft-link)
    wheel:            0.26.0-py27_1      --> 0.29.0-py27_0      (soft-link)
    zlib:             1.2.8-0            --> 1.2.8-3            (soft-link)

Proceed ([y]/n)? y

Fetching packages ...
libgfortran-3. 100% |################################| Time: 0:00:00 389.45 kB/s
mkl-11.3.3-0.t 100% |################################| Time: 0:00:33   3.76 MB/s
pixman-0.32.6- 100% |################################| Time: 0:00:01   1.32 MB/s
Extracting packages ...
[      COMPLETE      ]|###################################################| 100%
Unlinking packages ...
[      COMPLETE      ]|###################################################| 100%
Linking packages ...
An unexpected error has occurred, please consider sending the             |  31%
following traceback to the conda GitHub issue tracker at:

    https://github.com/conda/conda/issues

Include the output of the command 'conda info' in your report.


Traceback (most recent call last):
  File "/home/data/smortier/.conda/envs/PBM/bin/conda", line 5, in <module>
    sys.exit(main())
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 195, in main
    args_func(args, p)
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 202, in args_func
    args.func(args, p)
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/cli/main_install.py", line 45, in execute
    install.install(args, parser, 'install')
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/cli/install.py", line 423, in install
    plan.execute_actions(actions, index, verbose=not args.quiet)
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/plan.py", line 537, in execute_actions
    inst.execute_instructions(plan, index, verbose)
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/instructions.py", line 148, in execute_instructions
    cmd(state, arg)
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/instructions.py", line 91, in LINK_CMD
    link(state['prefix'], arg, index=state['index'])
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/instructions.py", line 87, in link
    install.link(pkgs_dir, prefix, dist, lt, index=index)
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/install.py", line 631, in link
    _link(src, dst, lt)
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/install.py", line 132, in _link
    shutil.copy2(src, dst)
  File "/home/software/anaconda/lib/python2.7/shutil.py", line 130, in copy2
    copyfile(src, dst)
  File "/home/software/anaconda/lib/python2.7/shutil.py", line 82, in copyfile
    with open(src, 'rb') as fsrc:
IOError: [Errno 13] Permission denied: u'/home/software/anaconda/pkgs/dbus-1.10.10-0/libexec/dbus-daemon-launch-helper'
```

output of `conda info`

```
Current conda install:

             platform : linux-64
        conda version : 3.18.5
  conda-build version : not installed
       python version : 2.7.10.final.0
     requests version : 2.8.1
     root environment : /home/software/anaconda  (read only)
  default environment : /home/data/smortier/.conda/envs/PBM
     envs directories : /home/data/smortier/.conda/envs
                        /home/data/smortier/envs
                        /home/software/anaconda/envs
        package cache : /home/data/smortier/.conda/envs/.pkgs
                        /home/data/smortier/envs/.pkgs
                        /home/software/anaconda/pkgs
         channel URLs : https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : None
    is foreign system : False
```

**The issue does not appear when going to sudo mode, sourcing the conda environment of the user and installing the package. The installation works just fine afterwards**

**Comment by mghijs** [comment]
The same permission error occurs for **matplotlib** (on the exact same system as Daan's):
```
conda install matplotlib                                                                                
Fetching package metadata: ....                                                                                                                 
Solving package specifications: ................................                                                                                
Package plan for installation in environment /home/data/mghijs/.conda/envs/PBM_plusTF:                                                          
                                                                                                                                                
The following NEW packages will be INSTALLED:                                                                                                   
                                                                                                                                                
    cycler:           0.10.0-py27_0     (soft-link)                                                                                             
    dbus:             1.10.10-0         (soft-link)                                                                                             
    fontconfig:       2.11.1-6          (soft-link)                                                                                             
    freetype:         2.5.5-1           (soft-link)                                                                                             
    glib:             2.43.0-1          (soft-link)                                                                                             
    gst-plugins-base: 1.8.0-0           (soft-link)                                                                                             
    gstreamer:        1.8.0-0           (soft-link)                                                                                             
    libpng:           1.6.22-0          (soft-link) 
    matplotlib:       1.5.3-np111py27_1                                                                                                         
    pip:              9.0.1-py27_0                                                                                                              
    pyparsing:        2.1.4-py27_0      (soft-link)                                                                                             
    pyqt:             5.6.0-py27_0      (soft-link)                                                                                             
    python-dateutil:  2.6.0-py27_0                                                                                                              
    pytz:             2016.7-py27_0     (soft-link)                                                                                             
    qt:               5.6.0-1           (soft-link)                                                                                             
    sip:              4.18-py27_0       (soft-link)                                                                                             
                                                                                                                                                
Proceed ([y]/n)? y                                                                                                                              
                                                                                                                                                
Linking packages ...                                                                                                                            
An unexpected error has occurred, please consider sending the                                                                             |   0%
following traceback to the conda GitHub issue tracker at:                                                                                       
                                                                                                                                                
    https://github.com/conda/conda/issues                                                                                                       
                                                                                                                                                
Include the output of the command 'conda info' in your report.                                                                                  
                                                                                                                                                
                                                                                                                                                
Traceback (most recent call last):                                                                                                              
  File "/home/data/mghijs/.conda/envs/PBM_plusTF/bin/conda", line 5, in <module>                                                                
    sys.exit(main())                                                                                                                            
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 195, in main                                               
    args_func(args, p)                                                                                                                          
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 202, in args_func                                          
    args.func(args, p)                                                                                                                          
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/cli/main_install.py", line 45, in execute                                     
    install.install(args, parser, 'install')                                                                                                    
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/cli/install.py", line 423, in install                                         
    plan.execute_actions(actions, index, verbose=not args.quiet)                                                                                
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/plan.py", line 537, in execute_actions                                        
    inst.execute_instructions(plan, index, verbose)                                                                                             
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/instructions.py", line 148, in execute_instructions                           
    cmd(state, arg)                                                                                                                             
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/instructions.py", line 91, in LINK_CMD                                        
    link(state['prefix'], arg, index=state['index'])                                                                                            
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/instructions.py", line 87, in link                                            
    install.link(pkgs_dir, prefix, dist, lt, index=index)                                                                                       
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/install.py", line 631, in link                                                
    _link(src, dst, lt)                                                                                                                         
  File "/home/software/anaconda/lib/python2.7/site-packages/conda/install.py", line 132, in _link                                               
    shutil.copy2(src, dst)                                                                                                                      
  File "/home/software/anaconda/lib/python2.7/shutil.py", line 130, in copy2                                                                    
    copyfile(src, dst)                                                                                                                          
  File "/home/software/anaconda/lib/python2.7/shutil.py", line 82, in copyfile                                                                  
    with open(src, 'rb') as fsrc:                                                                                                               
IOError: [Errno 13] Permission denied: u'/home/software/anaconda/pkgs/dbus-1.10.10-0/libexec/dbus-daemon-launch-helper'     
```

**Comment by kalefranz** [comment]
I can't really determine the exact cause here without a lot of back-and-forth or actually getting my hands on the machine.  However, I think we'll likely either (1) have this completely solved for you, or (2) at least be able to understand exactly what's happening, with the soon-forthcoming conda 4.3.

**Comment by DaanVanHauwermeiren** [comment]
@kalefranz there seems to be some issues with installing matplotlib 1.5.3 on our server. Version 1.5.1 installs and behaves correctly. Both in my case as in mghijs's case.
I'm not sure whether the issue lies in our server configuration or at mpl/conda.

**Comment by joerixaop** [comment]
@kalefranz The issue is actually less clear with conda 4.3, with conda 4.2 it complains about not being able to read a specific file (the dbus-daemon-launch-helper) for which you can then check the permissions (rwxr-x--- owned by root:root on my system), with conda 4.3 it only complains about "permission denied" without ever giving any details about what permissions are denied for.

Output with conda 4.2.9

```
e151286a@gdcrstoap701 ~> conda create -n my_root --clone=/opt/anaconda3/
Source:      /opt/anaconda3
Destination: /mnt/home/e151286a/.conda/envs/my_root
The following packages cannot be cloned out of the root environment:
 - conda-4.2.9-py35_0
 - conda-build-2.0.2-py35_0
Packages: 193
Files: 543
Linking packages ...
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   0%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   0%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   1%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   1%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   2%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   2%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   3%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   3%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   4%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   4%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   5%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   5%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   6%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   6%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   7%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   7%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   8%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   8%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   9%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |   9%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |  10%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |  10%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |  11%
WARNING conda.lock:touch(53): Failed to create lock, do not run conda in parallel processes [errno 13]                                                                                             |  11%
CondaOSError: OS error: failed to link (src='/opt/anaconda3/pkgs/dbus-1.10.10-0/libexec/dbus-daemon-launch-helper', dst='/mnt/home/e151286a/.conda/envs/my_root/libexec/dbus-daemon-launch-helper', type=3, error=PermissionError(13, 'Permission denied'))
```

Output with conda 4.3.6

```
e151286a@gdcrstoap701 ~> conda create -n my_root --clone=/opt/anaconda3/
Source:      /opt/anaconda3
Destination: /mnt/home/e151286a/.conda/envs/my_root
The following packages cannot be cloned out of the root environment:
 - conda-build-2.0.2-py35_0
 - conda-env-2.6.0-0
 - conda-4.3.8-py35_0
Packages: 193
Files: 541
ERROR conda.core.link:_execute_actions(318): An error occurred while installing package 'defaults::dbus-1.10.10-0'.
PermissionError(13, 'Permission denied')
Attempting to roll back.



PermissionError(13, 'Permission denied')
```

**Comment by joerixaop** [comment]
Seems to be the same root issue as #3614

**Comment by hannes-brt** [comment]
I have the same problem - I am unable to install matplotlib as a regular user in a conda env. Conda version is 4.3.14.

**Comment by thoangtrvn** [comment]
Here is my solution, for those who has root access; but only need to apply this once time; and it will work in a multi-user environment.

Suppose you install anaconda to /packages/anaconda3/3.4.1/

sudo chmod o+r /packages/anaconda3/4.3.1/pkgs/dbus-1.10.10-0/libexec/dbus-daemon-launch-helper
sudo chmod o+w /packages/anaconda3/4.3.1/conda-meta/ -R

**Comment by DaanVanHauwermeiren** [comment]
thanks @thoangtrvn for your solution proposal.

However I added the following commands: 
```
chmod a+r pkgs/ -R
chmod a+r bin -R
```
And it works fine it seems, installation goes smoothly.
I am aware that this is just a dirty workaround, but yeah, it works.

@kalefranz for completeness, this is my current conda install.
The conda version is > 4.3 and it still gave the same error.

```
Current conda install:

               platform : linux-64
          conda version : 4.3.8
       conda is private : False
      conda-env version : 4.3.8
    conda-build version : not installed
         python version : 2.7.12.final.0
       requests version : 2.13.0
       root environment : /home/software/anaconda  (read only)
    default environment : /home/software/anaconda
       envs directories : /home/data/dhauwe/.conda/envs
                          /home/software/anaconda/envs
          package cache : /home/data/dhauwe/.conda/envs/.pkgs
                          /home/software/anaconda/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/linux-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/linux-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/linux-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : None
           offline mode : False
             user-agent : conda/4.3.8 requests/2.13.0 CPython/2.7.12 Linux/2.6.32-642.15.1.el6.x86_64 CentOS/6.9 glibc/2.12
                UID:GID : 517:517
```

**Comment by DaanVanHauwermeiren** [comment]
This might be related to #1329

**Comment by OnlyBelter** [comment]
I install `anaconda` on Ubuntu by `sudo bash xxx.sh`,  we I install `spyder` by `conda`, I got same error: `ERROR conda.core.link:_execute_actions(335): An error occurred while installing package 'defaults::dbus-1.10.10-0'...`. Then I uninstalled anaconda, install it again without sudu, everything is ok. So I think if we want to install anaconda for all users, it would be a problem.

**Comment by kalefranz** [comment]
All permissions issues like these should be resolved as of conda 4.3.13.  Regarding the problem in #4838, I think it's probably incorrect permissions in the package itself, and we need to patch the package.

I'm going to close this ticket now, as any stack traces within are outdated and won't apply anymore.  If you come across this ticket with a similar problem for a conda version later than 4.3.13, please file a new ticket.
